File: BinarySearch/BinarySearch.c
#include <stdio.h>

int main()
{
    // array to be searched and variables
    int array[] = {1, 1, 3, 5, 6, 7, 9, 10};
    int target;
    int first = 0, middle, last = sizeof(array)/4;
    
    printf("Binary Search\n\n");
    
    // Prompt user for target value
    printf("What value do you want to find? ");
    scanf("%d", &target);
    
    // Calculate middle of array
    middle = (sizeof(array)/4)/2;
    
    // Loop to find the target value
    while (first<last){
        if (target > array[middle]){
            first = middle + 1;
            middle = (first + last)/2;
        }
        else{
            last = middle - 1;
            middle = (first + last)/2;
        }
        if (array[middle] == target){
            printf("The target is at index %d.\n",middle);
            return 0;
        }
    }
    
    // If the value was not found say so
    printf("The target is not in the array.\n");
    return 0;
}

File: BinarySearch/BinarySearch.py
def main():
    
    # Array of values to search through
    array = [1, 1, 3, 5, 6, 8, 9, 10]
    first = 0
    middle = int(len(array)/2)
    end = len(array)

    print("Binary Search\n")

    # Prompt user for target value
    target = int(input("What value do you want to find? "))

    # Binary search algorithm to find target in list
    while first <= end:
        
        if array[middle] == target:
            
            print("The target is at index {}.".format(middle))
            return
        
        elif target > array[middle]:
            
            first = middle
            middle = int((first + end)/2)
            
        else:
            
            end = middle
            middle = int((first + end)/2)

    # Indicate if the target is not in the array
    print("The target is not in the array.") 
    return


if __name__ == "__main__":
    main()
